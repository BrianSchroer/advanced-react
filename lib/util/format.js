export default class {
  static date(dateString) {
    return new Date(dateString).toDateString();
  }

  static dateTime(dateTimeString) {
    const dt = new Date(dateTimeString);
    return `${dt.toDateString()} ${dt.toLocaleTimeString([], {
      hour: '2-digit',
      minute: '2-digit'
    })}`;
  }
}
