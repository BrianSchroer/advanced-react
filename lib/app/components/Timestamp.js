import React from 'react';
import PropTypes from 'prop-types';
import format from '../../util/format';
import storeProvider from '../storeProvider';

export class Timestamp extends React.PureComponent {
  static propTypes = {
    timestampDisplay: PropTypes.string.isRequired
  };

  render() {
    return <div>{format.dateTime(this.props.timestampDisplay)}</div>;
  }
}

function extraProps(store) {
  return {
    timestampDisplay: format.dateTime(store.getState().timestamp)
  };
}

export default storeProvider(extraProps)(Timestamp);
