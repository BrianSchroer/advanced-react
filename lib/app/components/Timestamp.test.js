import React from 'react';
import { snapshotHelper } from '../../util/testHelpers';
import { Timestamp } from './Timestamp';

const testProps = {
  timestampDisplay: 'current time'
};

describe('Timestamp', () => {
  it('renders correctly', () => {
    snapshotHelper.assertMatch(<Timestamp {...testProps} />);
  });
});
