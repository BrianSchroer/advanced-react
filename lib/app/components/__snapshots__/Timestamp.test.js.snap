// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timestamp renders correctly 1`] = `
<div>
  Invalid Date Invalid Date
</div>
`;
